scom/stackstate_checks/scom/scom.py

```diff
@@ -82,114 +52,136 @@ def sendAlerts(self, session, scom_ip, domain, username, password, component_id)
-        # topology_instance_key = {"type": "scom", "url": scom_ip}
-        try:
-            self.start_snapshot()
-            for stream in streams:
-                data = {
-                    "fullClassName": "" + stream.get('class') + ""
-                }
-                response = (session.post('http://' + scom_ip + '/OperationsManager/data/scomObjectsByClass',
-                                         auth=HttpNtlmAuth(domain + '\\' + username, password),
-                                         json=data["fullClassName"])).json()
-                root_ids = []
-                root_tree = response.get("rows", [])
-                for component in root_tree:
-                    dataA = {"name": str(component.get("displayname")),
-                             "id": str(component.get("id")),
-                             "domain": str(stream.get('name')),
-                             "labels": [
-                                 str(stream.get('name'))
-                             ]
-                             }
-                    dataB = self.getComponentDetails(component.get("id"), session, scom_ip, domain, username, password)
-                    dataA.update(dataB)
-                    self.component(str(component.get("id")), str(stream.get('name') + "_component"), dataA)
-                    health = session.get(
-                        'http://' + scom_ip + '/OperationsManager/data/healthstate/' + component.get("id", None),
-                        auth=HttpNtlmAuth(domain + '\\' + username, password))
-                    self.event({
-                        "timestamp": int(time.time()),
-                        "msg_title": "Health Status",
-                        "msg_text": "Change in health status",
-                        "source_type_name": "Health",
-                        "host": component.get("displayname"),
-                        "tags": [
-                            "status:%s" % get_entity_health_state(session, scom_ip, domain, username, password,
-                                                                  health.text),
-                            "id:%s" % component.get("id")
-                        ]
-                    })
-                    self.sendAlerts(session, scom_ip, domain, username, password, component.get("id"))
-                    root_ids.append(component.get("id"))
-                self.serviceTree(root_ids, session, scom_ip, domain, username, password, stream.get('name'))
-            self.stop_snapshot()
-            session.close()
-        except Exception as e:
-            self.stop_snapshot()
-            self.log.exception("SCOM Error: %s" % str(e))
-            session.close()
-            self.service_check("scom", AgentCheck.CRITICAL, message=str(e))
-            raise e  # CheckException("SCOM Error: %s" % e), None, sys.exc_info()[2]
+        # read configuration file/yaml configuration
+        self.start_snapshot()
```

Comment:
Why are we not catching the exceptions anymore and sending in the service checks? I think you also need to stop the snapshot, unless of course we get an exception then we should not stop it

Reply:
added try except blocks